cms/djangoapps/contentstore/management/commands/import_content_library.py

```diff
@@ -0,0 +1,119 @@
+from xmodule.modulestore import ModuleStoreEnum
+from xmodule.modulestore.django import modulestore
+from xmodule.modulestore.exceptions import DuplicateCourseError
+from xmodule.modulestore.xml_importer import import_library_from_xml
+
+from cms.djangoapps.contentstore.utils import add_instructor
+from openedx.core.lib.extract_tar import safetar_extractall
+
+
+class Command(BaseCommand):
+    """
+    Import the specified content library archive.
+    """
+    help = 'Import the specified content library into mongo'
+
+    def add_arguments(self, parser):
+        parser.add_argument('archive_path')
+        parser.add_argument('owner_username')
+
+    def handle(self, *args, **options):
+        """
+        Given a content library archive path, import the corresponding course to mongo.
+        """
+
+        archive_path = options['archive_path']
+        username = options['owner_username']
+
+        data_root = Path(settings.GITHUB_REPO_ROOT)
+        subdir = base64.urlsafe_b64encode(os.path.basename(archive_path))
+        course_dir = data_root / subdir
+
+        # Extract library archive
+        tar_file = tarfile.open(archive_path)
+        try:
+            safetar_extractall(tar_file, course_dir.encode('utf-8'))
+        except SuspiciousOperation as exc:
+            raise CommandError(u'\n=== Course import {0}: Unsafe tar file - {1}\n'.format(archive_path, exc.args[0]))
+        finally:
+            tar_file.close()
+
+        # Paths to the library.xml file
+        abs_xml_path = os.path.join(course_dir, 'library')
+        rel_xml_path = os.path.relpath(abs_xml_path, data_root)
+
+        # Gather library metadata from XML file
+        xml_root = etree.parse(abs_xml_path / 'library.xml').getroot()
+        assert xml_root.tag == 'library'
+        metadata = xml_root.attrib
+        org = metadata['org']
+        library = metadata['library']
+        display_name = metadata['display_name']
+
+        # Fetch user and library key
+        user = User.objects.get(username=username)
+        courselike_key, created = _get_or_create_library(org, library, display_name, user)
+
+        # Check if data would be overwritten
+        ans = ''
+        while not created and ans.lower() not in ['y', 'yes', 'n', 'no']:
+            ans = raw_input(u'Library "{0}" already exists, overwrite it? [y/n] '.format(courselike_key))
+        if ans.startswith('n'):
+            print(u'Aborting import of "{0}"'.format(courselike_key))
+            return
+
+        # At last, import the library
+        try:
+            import_library_from_xml(
+                modulestore(), user.id,
+                settings.GITHUB_REPO_ROOT, [rel_xml_path],
+                load_error_modules=False,
+                static_content_store=contentstore(),
+                target_id=courselike_key
+            )
+        except Exception as e:
+            print(u'\n=== Failed to import library-v1:{0}+{1}'.format(org, library))
+            raise e
```

Comment:
A plain `raise` is better here because it won't change the traceback.

Reply:
Done.